src/utils/ethereumUtils.js

```diff
@@ -426,6 +445,78 @@ function openTransactionInBlockExplorer(hash, network) {
   Linking.openURL(`https://${etherscanHost}/tx/${normalizedHash}`);
 }
 
+async function parseEthereumUrl(data) {
+  let ethUrl;
+  try {
+    ethUrl = parse(data);
+  } catch (e) {
+    Alert.alert('Invalid ethereum url');
+    return;
+  }
+
+  const functionName = ethUrl.function_name;
+  let asset = null;
+  const network = getNetworkFromChainId(Number(ethUrl.chain_id || 1));
+  let address = null;
+  let nativeAmount = null;
+  const { nativeCurrency } = store.getState().settings;
+
+  while (store.getState().data.isLoadingAssets) {
+    await delay(300);
+  }
+  const { assets } = store.getState().data;
+
+  if (!functionName) {
+    // Send native asset
+    const nativeAssetAddress = getNativeAssetAddressForNetwork(network);
+    asset = getAsset(assets, toLower(nativeAssetAddress));
+
+    if (!asset || asset?.balance.amount === 0) {
+      Alert.alert(
+        'Ooops!',
+        `Looks like you don't have that asset in your wallet...`
+      );
+      return;
+    }
+    address = ethUrl.target_address;
+    nativeAmount = ethUrl.parameters?.value && fromWei(ethUrl.parameters.value);
+  } else if (functionName === 'transfer') {
+    // Send ERC-20
+    asset = getAsset(assets, toLower(ethUrl.target_address));
+    if (!asset || asset?.balance.amount === 0) {
+      Alert.alert(
+        'Ooops!',
+        `Looks like you don't have that asset in your wallet...`
+      );
+      return;
+    }
+    address = ethUrl.parameters?.address;
+    nativeAmount =
+      ethUrl.parameters?.uint256 &&
+      convertRawAmountToDecimalFormat(
+        ethUrl.parameters.uint256,
+        asset.decimals
+      );
+  } else {
+    Alert.alert('This action is currently not supported :(');
+    return;
+  }
+
+  const assetWithPrice = parseAssetsNative([asset], nativeCurrency)[0];
+
+  InteractionManager.runAfterInteractions(() => {
+    const params = { address, asset: assetWithPrice, nativeAmount };
+    if (isNativeStackAvailable || android) {
+      Navigation.handleAction(Routes.SEND_FLOW, {
+        params,
```

Comment:
I don't see we're using the network here, if the user gets a request for Optimism the asset would be ETH, would the user know / send the tx on optimism or mainnet?

Reply:
We get the network from the chainId, and then we request the asset for that network https://github.com/rainbow-me/rainbow/pull/2343/files#diff-34ca43434805a064f0600ef0ff5a7dbec13f0ae719ba1c19afeffe0fbd1d9b3fR459